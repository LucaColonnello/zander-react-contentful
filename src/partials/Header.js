import React from 'react'
import parseDomain from 'parse-domain'
import months from '../utils/months'

const Header = (props) => {
  let formattedDate = ''
  if (props.date) {
    formattedDate = new Date(props.date)
  }

  let formattedUrl = ''
  if (props.originalUrl) {
    formattedUrl = `${parseDomain(props.originalUrl).domain}.${parseDomain(props.originalUrl).tld}`
  }

  return (
    <header className="header l-container header--article">
      <div className="header-inner">
        <h1 className="header-title p-name" itemProp="headline">{props.title}</h1>

        {
          props.subtitle &&
          <h2 className="header-subtitle p-summary">{props.subtitle}</h2>
        }

        {
          props.date &&
          <div className="header-meta">
            <time dateTime={props.date} className="dt-published">{formattedDate.getDay()} {months[formattedDate.getMonth()]} {formattedDate.getUTCFullYear()} </time>
            {
              props.originalUrl &&
              <span>
                &bull; Originally posted at <a href={props.originalUrl}>{formattedUrl}</a>
              </span>
            }
            <meta itemProp="datePublished" content={props.date} />
          </div>
        }

      </div>
      <hr className="hr--short" />
    </header>
  )
}

export default Header
