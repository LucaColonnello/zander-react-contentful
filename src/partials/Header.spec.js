import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from './Header';

test('Header snapshot test', () => {
  const component = shallow(
    <Header
      title="The title"
      subtitle="The subtitle"
      body="The body"
      date="2017-02-02T00:00+00:00"
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
