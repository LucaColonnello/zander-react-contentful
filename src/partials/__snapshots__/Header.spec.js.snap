// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
<header
  className="header l-container header--article"
>
  <div
    className="header-inner"
  >
    <h1
      className="header-title p-name"
      itemProp="headline"
    >
      The title
    </h1>
    <h2
      className="header-subtitle p-summary"
    >
      The subtitle
    </h2>
    <div
      className="header-meta"
    >
      <time
        className="dt-published"
        dateTime="2017-02-02T00:00+00:00"
      >
        4
         
        Feb
         
        2017
         
      </time>
      <meta
        content="2017-02-02T00:00+00:00"
        itemProp="datePublished"
      />
    </div>
  </div>
  <hr
    className="hr--short"
  />
</header>
`;
